src/lib/utils/settings/log_level_setting.cpp

```diff
@@ -0,0 +1,25 @@
+#include "log_level_setting.hpp"
+
+#include "constant_mappings.hpp"
+#include "hyrise.hpp"
+
+namespace opossum {
+
+LogLevelSetting::LogLevelSetting(const std::string& init_name) : AbstractSetting(init_name) {}
+
+const std::string& LogLevelSetting::description() const {
+  static const auto description = std::string{"Minimal log level of the LogManager"};
+  return description;
+}
+
+const std::string& LogLevelSetting::get() { return log_level_to_string.left.at(Hyrise::get().log_manager._log_level); }
+
+void LogLevelSetting::set(const std::string& value) {
+  Hyrise::get().log_manager._log_level = log_level_to_string.right.at(value);
```

Comment:
This will kill everything if I mistyped the log level, right? We don't want AssertInput all over the code base, but maybe we need something more lenient here.

Reply:
I was something I was not sure about, I'm gonna add this

Reply:
Could be something to discuss in a more general way. Other settings will see the same.

Reply:
For now, I simply checked if the value is ok. One could also add a try_set method to AbstractSetting that catches errors.